### DISCUSSION TITLE
For the display of more than 25 different LORAs

### DISCUSSION THREAD

**Discussion by AI-Avalon** [Kilvoctu/aiyabot#190] [Ideas]
**Describe the bug**  
A clear and concise description of what the bug is. Please keep each bug report limited to one issue.
I can't display more than 25 selective arguments due to Discord's specifications, but I would like to see some way to improve the display when more than 25 types of Lora are put in.
---

**Environment**
 - OS: [e.g. Windows / Linux]
Windows
---

**Console logs**  
There were no errors.

**Comment by Kilvoctu** [comment]
Heya, this sounds like a suggestion, rather than a bug. I'd like "Issues" to be for bug reports only, so gonna move this to Discussions.  

Can you explain some ways you'd like the LoRAs to be displayed?

**Reply by AI-Avalon** [reply]
It may indeed be a suggestion. Apologies.

If you do not know the number of models (in this case, we assume multiple models such as D checkpoint models, Lora models, etc.) or if there are more than 25 models, you can divide them into pages (Lora-page1, Lora-page2, etc.) to display model names without getting caught in the display limit. If you have more than 25 models

**Reply by AI-Avalon** [reply]
As a side note, here's how I envision it
data_model
in the lower layer of
Check_point_1
Check_point_2
Check_point_3
It would be nice to be able to select and increase the number of items like this.

**Reply by Kilvoctu** [reply]
> If you do not know the number of models (in this case, we assume multiple models such as D checkpoint models, Lora models, etc.) or if there are more than 25 models, you can divide them into pages (Lora-page1, Lora-page2, etc.) to display model names without getting caught in the display limit. If you have more than 25 models

The /info command already informs users of what is available.
<details>
<summary>example</summary>
<br>

![image](https://github.com/Kilvoctu/aiyabot/assets/2993060/c0c2a00b-6514-4676-a403-3e654a5f6b6e)

</details>


If you mean directly into the /draw slash command, then that's not possible as far as I know. Discord has a hard limit of 25 items for its selection popups. I already am using a workaround ([discord.AutocompleteContext](https://docs.pycord.dev/en/stable/api/application_commands.html#discord.AutocompleteContext)) to exceed the limit, but it's still limited to displaying 25 items. If you know any other workaround, please share.
